product-base/makefile

```
@@ -7,11 +7,18 @@ TAG = zenoss/$(IMAGENAME):$(VERSION)_$(BUILD_NUMBER)_$(MATURITY)
 
 .PHONY: build push clean
 
-build:
+build:  component_versions
 	docker build --no-cache=true -t $(TAG) .
 
 push:
 	docker push $(TAG)
 
 clean:
+	rm -rf component_info
 	-docker rmi -f $(TAG)
+
+component_versions: component_info 
+	@cp ../artifact_download.py ../component_versions.json component_info/.
+
+component_info:
+	@mkdir component_info 
```

I could be wrong, but I'm guessing if you `make build` two times in a row without a clean in btwn then the second make will fail on the `mkdir` at line 24 will error out because the component_info already exists. You can put a minus '`-`' in front of the mkdir command to tell make to ignore errors for that recipe